src/ifcopenshell-python/ifcopenshell/util/unit.py

```diff
@@ -445,56 +451,83 @@ def get_property_unit(
     :return: The IFC unit entity, or nothing if there is no default project
         unit defined.
     """
-    unit = getattr(prop, "Unit", None)
-    if unit:
+    unit = prop.Unit
```

Comment:
NOTE: I had considered leaving this `getattr(prop, "Unit", None)`, thinking this might be "safer". However even in the original code one often just writes `prop.UpperBoundValue` assuming the `__get_attr__` magic will just work, so for consistency I applied the same style.

If there be some particular reason to assume this is unsafe, we can definitely revert this to `getattr(prop, "Unit", None)`. Though in this case, we should probably then do the same for the other `prop. ...`-methods.